subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/DependencyHandlerScope.kt

```diff
@@ -224,6 +226,31 @@ private constructor(
     inline operator fun <T : ModuleDependency> Configuration.invoke(dependency: T, dependencyConfiguration: T.() -> Unit): T =
         add(name, dependency, dependencyConfiguration)
 
+    /**
+     * Adds a dependency provider to the given configuration.
+     *
+     * @param dependency the dependency provider to be added.
+     * @param dependencyConfiguration the configuration to be applied to the dependency
+     *
+     * @see [DependencyHandler.addProvider]
+     * @since 6.9
+     */
+    @Incubating
+    operator fun Configuration.invoke(dependency: Provider<*>, dependencyConfiguration: ExternalModuleDependency.() -> Unit) =
```

Comment:
:x: `operator fun String.invoke(...)` overloads are needed for DSL symmetry

Reply:
Oh for `"implementation"(...) { ... }` ?

Reply:
Yes!

Reply:
done